dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/inbound/HttpInboundLink.java

```diff
@@ -942,51 +938,45 @@ public boolean isHTTP2UpgradeRequest(Map<String, String> headers) {
      * @param headers a String map of http header key and value pairs
      * @return true if an http2 upgrade header is found
      */
-    private boolean checkIfUpgradeHeaders(Map<String, String> headers) {
+    private boolean checkIfUpgradeHeaders(Enumeration<String> connection, Enumeration<String> upgrade) {
         // looking for two headers.
         // connection header with a value of "upgrade"
         // upgrade header with a value of "h2c"
         boolean connection_upgrade = false;
         boolean upgrade_h2c = false;
-        String headerValue = null;
-        Set<Entry<String, String>> headerEntrys = headers.entrySet();
-
-        for (Entry<String, String> header : headerEntrys) {
-            String name = header.getKey();
+        while (connection.hasMoreElements()) {
```

Comment:
Based off of other comments that I talked about instead of doing entrySet you can just call two headers.get() methods instead.  Should be much simpler and perform better than the old impl.